### PR TITLE
feat: Add partial pause only for send operation

### PR DESCRIPTION
**Pull request by gfournieriExec**

### CODE REVIEW

File: src/IexecLayerZeroBridge.sol [hunk 1]
```diff
@@ -166,14 +181,18 @@ contract IexecLayerZeroBridge is
...
      *
-     * @custom:security Only callable when contract is not paused
+     * @custom:security Uses whenEntrancesNotPaused which checks both pause levels
      * @custom:security Requires the RLC token to have granted burn permissions to this contract
      */
     function _debit(address _from, uint256 _amountLD, uint256 _minAmountLD, uint32 _dstEid)
         internal
         override
-        whenNotPaused
+        whenEntrancesNotPaused // This checks both full pause and entrance pause
```
Comment (zguesmi) [inline src/IexecLayerZeroBridge.sol:195]: Why not?
```suggestion
        whenNotPaused whenEntriesNotPaused
```

Comment (gfournieriExec) [reply src/IexecLayerZeroBridge.sol:195]: `whenEntriesNotPaused` checks that `whenNotPaused` is not set checks `_requireEntrancesNotPaused` implem
maybe that could be confusing for the user but I guess the documentation is enough

File: src/DualPausableUpgradeable.sol [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        return $._entrancesPaused;
+    }
+
+    /**
+     * @dev Returns the overall operational state
+     * @return fullyPaused True if complete pause is active
+     * @return entrancesPaused_ True if entrance pause is active
+     * @return fullyOperational True if neither pause is active
+     */
+    function pauseState()
```
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply src/DualPausableUpgradeable.sol:105]: I feel that the meaning of `isPaused()` can be confusing whit `paused`()

File: src/IexecLayerZeroBridge.sol (hunk 1 above)
Comment (gfournieriExec) [reply src/IexecLayerZeroBridge.sol:195]: ok gonna change that

File: test/units/IexecLayerZeroBridge.t.sol [hunk 3]
```diff
@@ -152,165 +152,132 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
         assertEq(rlcCrosschainToken.balanceOf(user1), INITIAL_BALANCE - TRANSFER_AMOUNT);
     }
 
-    // ============ LEVEL 2 PAUSE TESTS (Entrance Pause) ============
+    // ============ LEVEL 2 PAUSE TESTS (Send Pause) ============
 
-    function test_PauseEntrances_EmitsCorrectEvent() public {
+    function test_PauseSends_EmitsCorrectEvent() public {
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply test/units/IexecLayerZeroBridge.t.sol:157]: This is not part of the bridge tests. I would've create a file `DualPausableUPgradeable.t.sol` and moved this test (and related ones) there.

File: test/units/IexecLayerZeroBridge.t.sol [hunk 4]
```diff
@@ -112,9 +148,137 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+        // Should now work normally
+        assertFalse(iexecLayerZeroBridge.paused());
+        assertFalse(iexecLayerZeroBridge.sendPaused());
+
+        test_SendToken_WhenOperational();
+    }
+
+    // ============ DUAL PAUSE WORKFLOW TESTS ============
+
+    function test_DualPause_PauseFromSendToFull() public {
```
Comment (zguesmi) [inline test/units/IexecLayerZeroBridge.t.sol:213]: This test and an additional one `test_PauseFromFullToSendOnly` should be inside `DualPausableUPgradeable.t.sol`.

File: test/units/IexecLayerZeroBridge.t.sol [hunk 5]
```diff
@@ -112,9 +148,137 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+        emit Paused(pauser);
+
+        iexecLayerZeroBridge.pause();
+        vm.stopPrank();
+
+        assertTrue(iexecLayerZeroBridge.paused());
+        assertTrue(iexecLayerZeroBridge.sendPaused());
+    }
+
+    function test_PauseStatus_ReturnsCorrectStates() public {
```
Comment (zguesmi) [inline test/units/IexecLayerZeroBridge.t.sol:229]: Same, in `DualPausableUPgradeable.t.sol`.

File: test/units/IexecLayerZeroBridge.t.sol [hunk 6]
```diff
@@ -112,9 +148,137 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+        iexecLayerZeroBridge.pause();
+
+        (fullyPaused, onlySendPaused) = iexecLayerZeroBridge.pauseStatus();
+        assertTrue(fullyPaused);
+        assertTrue(onlySendPaused);
+    }
+
+    // ============ EDGE CASE TESTS ============
+
+    function test_PauseSends_CannotUnpauseWhenNotPaused() public {
```
Comment (zguesmi) [inline test/units/IexecLayerZeroBridge.t.sol:254]: Same thing for all of these tests.

File: test/units/RLCAdapter.t.sol [hunk 7]
```diff
@@ -115,6 +159,136 @@ contract RLCAdapterTest is TestHelperOz5 {
...
+        // Should now work normally
+        assertFalse(adapter.paused());
+        assertFalse(adapter.sendPaused());
+
+        test_SendToken_WhenOperational();
+    }
+
+    // ============ DUAL PAUSE WORKFLOW TESTS ============
+
+    function test_DualPause_PauseFromSendToFull() public {
```
Comment (zguesmi) [inline test/units/RLCAdapter.t.sol:225]: Same thing for the below tests (they are redundant anyway).

File: src/DualPausableUpgradeable.sol
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/DualPausableUpgradeable.sol:1]: And why not bridges/common? Since it doesn’t concern token logic?
